### PR TITLE
Hello Rails React As One App

### PR DESCRIPTION
**Pull request by paulsaenzsucre**
## Hello Rails React As One App

The proposed changes for this pull request are:

:pushpin: We implement a rails app using react as front end library. and postgreSQL as a database

:pushpin: We implement a table for storing messages and and seed with different greetings.

:pushpin: We implement an API endpoint that selects a random greeting from the table through a controller with an action and Rails route.

:pushpin: We implement a static view that it's the app's root.

:pushpin: We implement the App component with react-router and render it in our static view.

:pushpin: We implement the Greeting component that display a greeting. We set it up as a route in our App component.

:pushpin: We implement a store, an action, and a reducer that connect to our API endpoint to get the random greeting.

:pushpin: We display the random greeting in our Greeting component.

image

### CODE REVIEW

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @paulsaenzsucre, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that lacks a link to the `react` repo in your readme. Please add the link to the `frontend` repo in the backend `readme` file and the link to the `backend` in the `frontend` readme file. Make sure the `frontend` uses a professional `readme` template like in the `backend`.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

## Find me on slack
### Martin Afani

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by paulsaenzsucre** [comment]
@afanimartin Thank you for your review, I tried to contact you on slack. I think you confused about projects, I submitted "Hello Rails React as One App" but you required me about features of another project. Don't worry, I will submit it again.

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Rails.application.routes.draw do
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+  # Defines the root path route ("/")
+  # root "articles#index"
```
Comment (t-yanick) [inline config/routes.rb]: - [ ] Great job working on this milestone. Kindly remove this commented line of code and any other in your codebase. It is not professional to keep commented pieces of code in our codebase.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @paulsaenzsucre , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Required gem used and needed dependencies added
:heavy_check_mark: Database and API endpoint created
:heavy_check_mark: Required components added
:heavy_check_mark: Apps work as required

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**